flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java

```diff
@@ -121,6 +124,23 @@ public WindowedStream(KeyedStream<T, K> input,
 	}
 
 	/**
+	 * Sets the allowed lateness. If the {@link WindowAssigner} used
+	 * is in processing time, then the allowed lateness is set to 0.
+	 */
+	@PublicEvolving
+	public WindowedStream<T, K, W> setAllowedLateness(Time lateness) {
+		long millis = lateness.toMilliseconds();
+		if (allowedLateness < 0) {
+			throw new IllegalArgumentException("The allowed lateness cannot be negative.");
+		} else if (allowedLateness != 0 && !windowAssigner.isEventTime()) {
+			this.allowedLateness = 0;
```

Comment:
Yeah, but it would throw right when specifying the job. Not when running or anything. LOG is also fine, though. 😃